src/azure-cli/azure/cli/command_modules/rdbms/_util.py

```diff
@@ -77,9 +78,25 @@ def create_firewall_rule(cmd, resource_group_name, server_name, start_ip, end_ip
     firewall_client = cf_postgres_firewall_rules(cmd.cli_ctx, None)
     if db_engine == 'mysql':
         firewall_client = cf_mysql_firewall_rules(cmd.cli_ctx, None)
+        parameters = mysql.models.FirewallRule(
+            name=firewall_name,
+            start_ip_address=start_ip,
+            end_ip_address=end_ip
+        )
     elif db_engine == 'mariadb':
         firewall_client = cf_mariadb_firewall_rules(cmd.cli_ctx, None)
+        parameters = mariadb.models.FirewallRule(
+            name=firewall_name,
+            start_ip_address=start_ip,
+            end_ip_address=end_ip
+        )
+    else:
+        parameters = postgresql.models.FirewallRule(
+            name=firewall_name,
+            start_ip_address=start_ip,
+            end_ip_address=end_ip
+        )
```

Comment:
```suggestion
    if db_engine == 'mysql':
        firewall_client = cf_mysql_firewall_rules(cmd.cli_ctx, None)
    elif db_engine == 'mariadb':
        firewall_client = cf_mariadb_firewall_rules(cmd.cli_ctx, None)
    parameters = {'name': firewall_name, 'start_ip_address': start_ip, 'end_ip_address': end_ip}
```

Since all `FirewallRule`s inherit `msrest.serialization.Model`, you can define plain `dict` and it will be serialized automatically. This will make your code cleaner.

Reply:
Thanks